### PR TITLE
added translations of core.md

### PR DESCRIPTION
**Pull request by miyamotok0105**
core.mdの翻訳を行いました。なるべく文言を統一しました。確認お願いします。

### CODE REVIEW

File: sources/layers/core.md [hunk 1]
```diff
@@ -2,59 +2,63 @@
 ### Dense
 
 ```python
-keras.layers.core.Dense(output_dim, init='glorot_uniform', activation='linear', weights=None, W_regularizer=None, b_regularizer=None, activity_regularizer=None, W_constraint=None, b_constraint=None, bias=True, input_dim=None)
+keras.layers.core.Dense(units, activation=None, use_bias=True, kernel_initializer='glorot_uniform', bias_initializer='zeros', kernel_regularizer=None, bias_regularizer=None, activity_regularizer=None, kernel_constraint=None, bias_constraint=None)
 ```
 
 通常の全結合ニューラルネットワークレイヤー．
 
+`Dense`実装する操作：`output = activation(dot(input, kernel) + bias)`ここで,`activation`は`activation`引数として渡される要素単位の起動機能で,`kernel`はレイヤーによって作成された重み行列であり,`bias`はレイヤーによって作成されたバイアスベクトルです.（`use_bias`が`True`の場合にのみ適用されます）.
```
Comment (nzw0301) [inline sources/layers/core.md]: - `Dense`が実行する操作 のほうがわかりやすいかなと思うのですが，どうでしょうか？
- Activation Functionは活性化関数 (not 起動機能)

File: sources/layers/core.md [hunk 2]
```diff
@@ -2,59 +2,63 @@
...
 ```
 
 __引数__
 
-- __output_dim__: 正の整数 > 0.
-- __init__: レイヤーの重みに対する初期化関数名([initializations](../initializations.md)を参照)，もしくは，重みを初期化するために使用するTheano関数．このパラメータは`weights`引数を与えていないときにのみ有効です．
+- __units__:正の整数,出力空間の次元数
 - __activation__: 使用する活性化関数名
-	([activations](../activations.md)を参照)，もしくは，要素ごとのTheano関数．
+	([activations](https://keras.io/activations/)を参照)
```
Comment (nzw0301) [inline sources/layers/core.md]: リンク修正は不要かと思うのですが，いかがでしょうか？

File: sources/layers/core.md [hunk 3]
```diff
@@ -2,59 +2,63 @@
...
+# after the first layer, you don't need to specify
+# the size of the input anymore:
 model.add(Dense(32))
 ```
 
 __引数__
 
-- __output_dim__: 正の整数 > 0.
-- __init__: レイヤーの重みに対する初期化関数名([initializations](../initializations.md)を参照)，もしくは，重みを初期化するために使用するTheano関数．このパラメータは`weights`引数を与えていないときにのみ有効です．
+- __units__:正の整数,出力空間の次元数
```
Comment (nzw0301) [inline sources/layers/core.md]: `,`は全角の`，`でお願いします　
以下すべての日本語訳中の `,` についても同様

File: sources/layers/core.md [hunk 4]
```diff
@@ -2,59 +2,63 @@
...
-- __activity_regularizer__: ネットワーク出力に適用される[ActivityRegularizer](../regularizers.md)のインスタンス．
-- __W_constraint__: 主の重み行列に適用される[constraints](../constraints.md)モジュールのインスタンス．(例えば，maxnorm，nonneg)．
-- __b_constraint__: バイアスに適用される[constraints](../constraints.md)モジュールのインスタンス．
-- __bias__: バイアスを含めるかどうか(すなわち，線形というよりもむしろアフィンレイヤーにさせるか)．
-- __input_dim__: 入力の次元(整数)．
-	この引数(もしくは，キーワード引数`input_shape`)
-	は，モデルの最初のレイヤーとして使うときに必要とされる．
+	(すなわち，"線形"活性化: `a(x) = x`)．
+- __use_bias__: レイヤーがバイアスベクトルを使用するかどうか．
+- __kernel_initializer__: `kernel`重み行列の初期化子([initializations](https://keras.io/initializers/)を参照)
```
Comment (nzw0301) [inline sources/layers/core.md]: 以下リンクと`子`について共通したコメント: 

- `子`はなぜあるのでしょうか？
- リンクは日本語のページへ移動するようにして下さい

File: sources/layers/core.md [hunk 5]
```diff
@@ -2,59 +2,63 @@
...
+- __bias_initializer__: バイアスベクトルの初期化子([initializations](https://keras.io/initializers/)を参照)
+- __kernel_regularizer__: `kernel`重み行列に適用される正規化関数([regularizers](https://keras.io/regularizers/)を参照)
```
Comment (nzw0301) [inline sources/layers/core.md]: 以下 `正規化` について共通したコメント: 

`regularizer`の訳は正則化でお願いします（目次と共通にするため）

File: sources/layers/core.md [hunk 6]
```diff
@@ -2,59 +2,63 @@
...
+- __bias_regularizer__: バイアスベクトルに適用される正則化関数([regularizers](https://keras.io/regularizers/)を参照)
+- __activity_regularizer__: レイヤーの出力に適用される正規化関数(activation)([regularizers](https://keras.io/regularizers/)を参照)
+- __kernel_constraint__: `kernel`重み行列に適用される制約関数([constraints](https://keras.io/constraints/)を参照)
+- __bias_constraint__: バイアスベクトルに適用される制約関数([constraints](https://keras.io/constraints/)を参照)
+
 
 __入力の型__
 
-2次元テンソルの型: `(nb_samples, input_dim)`．
+n次元テンソルの型: `(batch_size, ..., input_dim)`．
```
Comment (nzw0301) [inline sources/layers/core.md]: 以下共通
- `:`→`：`

File: sources/layers/core.md [hunk 7]
```diff
@@ -66,39 +70,41 @@ keras.layers.core.Activation(activation)
...
-<span style="float:right;">[[source]](https://github.com/fchollet/keras/blob/master/keras/layers/core.py#L67)</span>
+<span style="float:right;">[[source]](https://github.com/fchollet/keras/blob/master/keras/layers/core.py#L72)</span>
 ### Dropout
 
 ```python
-keras.layers.core.Dropout(p)
+keras.layers.core.Dropout(rate, noise_shape=None, seed=None)
 ```
 
 入力にドロップアウトを適用する．ドロップアウトは，訓練時のそれぞれの更新において入力ユニットの`p`をランダムに0にセットすることであり，それは過学習を防ぐのを助ける．
```
Comment (nzw0301) [inline sources/layers/core.md]: - `p` → `rate`

File: sources/layers/core.md [hunk 8]
```diff
@@ -66,39 +70,41 @@ keras.layers.core.Activation(activation)
...
 
 __引数__
 
-- __p__: 0と1の間の浮動小数点数．入力ユニットをドロップする割合．
+- __rate__: 0と1の間の浮動小数点数．入力ユニットをドロップする割合．
+- __noise_shape__: バイナリドロップアウトマスクの型の1次元の整数テンソルが入力に乗算される．例えば入力の型を`(batch_size, timesteps, features)`とし,ドロップアウトマスクをすべてのタイムステップで同じにしたい場合,`noise_shape=(batch_size, 1, features)`を使うことができる.
```
Comment (nzw0301) [inline sources/layers/core.md]: 乗算するのはマスクで，1Dテンソルはmaskのshape(:noise_shape)だと思います

File: sources/layers/core.md [hunk 9]
```diff
@@ -2,59 +2,63 @@
...
-keras.layers.core.Dense(output_dim, init='glorot_uniform', activation='linear', weights=None, W_regularizer=None, b_regularizer=None, activity_regularizer=None, W_constraint=None, b_constraint=None, bias=True, input_dim=None)
+keras.layers.core.Dense(units, activation=None, use_bias=True, kernel_initializer='glorot_uniform', bias_initializer='zeros', kernel_regularizer=None, bias_regularizer=None, activity_regularizer=None, kernel_constraint=None, bias_constraint=None)
 ```
 
 通常の全結合ニューラルネットワークレイヤー．
 
+`Dense`実装する操作：`output = activation(dot(input, kernel) + bias)`ここで,`activation`は`activation`引数として渡される要素単位の起動機能で,`kernel`はレイヤーによって作成された重み行列であり,`bias`はレイヤーによって作成されたバイアスベクトルです.（`use_bias`が`True`の場合にのみ適用されます）.
+
+
+- Note：レイヤーへの入力のランクが2より大きい場合は,`kernel`を使用した最初のドット積の前にフラット化されます.
```
Comment (nzw0301) [inline sources/layers/core.md]: フラット化 → 平坦化

File: sources/layers/core.md [hunk 10]
```diff
@@ -112,16 +118,16 @@ __例__
 
 ```python
 model = Sequential()
-model.add(Convolution2D(64, 3, 3, border_mode='same', input_shape=(3, 32, 32)))
-# いま: model.output_shape == (None, 64, 32, 32)
+model.add(Conv2D(64, (3, 3) ,input_shape=(3, 32, 32)))
```
Comment (nzw0301) [inline sources/layers/core.md]: ','の位置がおかしいです： (3, 3) , -> (3, 3),

File: sources/layers/core.md [hunk 11]
```diff
@@ -132,8 +138,7 @@ keras.layers.core.Reshape(target_shape)
 
 __引数__
 
-- __target_shape__: ターゲットの型．整数のタプル，
-	サンプルの次元を含まない(バッチサイズ)．
+- __target_shape__: ターゲットの型．整数のタプル，サンプルの次元を含まない(バッチサイズ)．
```
Comment (nzw0301) [inline sources/layers/core.md]: 以下共通: 翻訳文の`(`と`)`は全角でお願いします

File: sources/layers/core.md [hunk 12]
```diff
@@ -364,7 +338,7 @@ keras.layers.core.Masking(mask_value=0.0)
 入力テンソル(テンソルの次元 #1)のそれぞれのタイムステップに対して，
 もしそのタイムステップの入力テンソルのすべての値が`mask_value`に等しいなら，そのときそのタイムステップはすべてのダウンストリームレイヤー(それらがマスキングをサポートしている限り)でマスク(スキップ)されるでしょう．
 
-もしどんなダウンストリームレイヤーが，そのような入力マスクをまだ受けておらず，マスキングをサポートしていなければ，例外が生じるだろう．
+もしどんな下流レイヤーが，そのような入力マスクをまだ受けておらず，マスキングをサポートしていなければ，例外が生じるだろう．
```
Comment (nzw0301) [inline sources/layers/core.md]: `生じるだろう` →`生じます`

File: sources/layers/core.md [hunk 13]
```diff
@@ -364,7 +338,7 @@ keras.layers.core.Masking(mask_value=0.0)
 入力テンソル(テンソルの次元 #1)のそれぞれのタイムステップに対して，
 もしそのタイムステップの入力テンソルのすべての値が`mask_value`に等しいなら，そのときそのタイムステップはすべてのダウンストリームレイヤー(それらがマスキングをサポートしている限り)でマスク(スキップ)されるでしょう．
```
Comment (nzw0301) [inline sources/layers/core.md]: 下流レイヤーに統一？

**Comment by nzw0301** [comment]
L349とL350の訳をお願いします

File: sources/layers/core.md (hunk 1 above)
Comment (Wakamatz) [reply sources/layers/core.md]: 「ここで 」より「ただし」の方が日本語の表現としては妥当だと思います．
下記のような感じでどうでしょう．

`Dense`が実装する操作：`output = activation(dot(input, kernel) + bias)`．ただし，～

File: sources/layers/core.md (hunk 12 above)
Comment (Wakamatz) [reply sources/layers/core.md]: この場合の「yet」は接続詞で「それにもかかわらず」と訳すのが妥当だと思います．
それから「will」は人間の意思が介在しない場合は現在形で訳すほうが自然な気がします．
下記のような感じでいかがでしょうか？

> 下流レイヤーがマスキングをサポートしていないのにそのような入力マスクを受け取ると例外が発生します．

File: sources/layers/core.md (hunk 9 above)
Comment (ynqa) [reply sources/layers/core.md]: [nits] `平滑化` がいいと思います。

**Comment by nzw0301** [comment]
L114 `平坦化`から`平滑化`

File: sources/layers/core.md (hunk 9 above)
Comment (Wakamatz) [reply sources/layers/core.md]: `kernel`とのドット積を可能にするために入力を1Dテンソルに変換することを"be flattend"と言ってると思うので，平滑化は適切では無いと思います．
フラット化のままで良いと思いますが，いかがでしょうか？

Comment (nzw0301) [reply sources/layers/core.md]: やっていることは`Flatten`と同じくサンプル軸を以外を1Dテンソルとみなしてドット積とってるので，`Flatten`の説明と揃えたほうがいいかなと思っています．
一方で，https://github.com/tensorflow/tensorflow/pull/9043#issuecomment-293073944
で言われてるように `flatten`してから `Dense`するわけではないので，もう少し補った説明にした方がわかりやすいのかもしれません

File: sources/layers/core.md (hunk 13 above)
Comment (miyamotok0105) [reply sources/layers/core.md]: 深層学習の日本語でダウンストリームより下流の方が使われている気がするのですが如何でしょうか。

Comment (nzw0301) [reply sources/layers/core.md]: 個人的に下流とダウンストリームどちらでも意味がよくわからないので補って書いたほうがいいかなと感じています．
ただ下流という表現がよく使われているのでしたら問題ないかと思います

Comment (Wakamatz) [reply sources/layers/core.md]: 下流の方が意味が取りやすいので下流に賛同します．

File: sources/layers/core.md (hunk 9 above)
Comment (ynqa) [reply sources/layers/core.md]: > Flattenと同じくサンプル軸を以外を1Dテンソルとみなしてドット積とってるので，Flattenの説明と揃えたほうがいいかなと思っています

自分もその認識でコメントさせていただきました 🙏 
安易にコメントを書いてしまってすみません。

> flattenしてから Denseするわけではない

ただ、確かにこの通りの挙動でした。なので、同じく補足的な説明があれば（どう訳すかとは別に）よりわかりやすくなると思いました！

[Ref] https://github.com/tensorflow/tensorflow/issues/8175

File: sources/layers/core.md (hunk 1 above)
Comment (miyamotok0105) [reply sources/layers/core.md]: Denseが実装する操作：output = activation(dot(input, kernel) + bias)．ただし，～に修正致しました。

Comment (miyamotok0105) [reply sources/layers/core.md]: 起動機能→活性化関数に修正致しました。

File: sources/layers/core.md (hunk 2 above)
Comment (miyamotok0105) [reply sources/layers/core.md]: リンク全般を日本語ページのリンクに修正致しました。

File: sources/layers/core.md (hunk 3 above)
Comment (miyamotok0105) [reply sources/layers/core.md]: 日本語部分を全般的に，に変更致しました。

File: sources/layers/core.md (hunk 4 above)
Comment (miyamotok0105) [reply sources/layers/core.md]: 子を削除致しました。
リンクを日本語ページに変更いたしました。

File: sources/layers/core.md (hunk 5 above)
Comment (miyamotok0105) [reply sources/layers/core.md]: 正規化を正則化に修正致しました。

File: sources/layers/core.md (hunk 6 above)
Comment (miyamotok0105) [reply sources/layers/core.md]: 全般的に:→：に修正致しました。

File: sources/layers/core.md (hunk 7 above)
Comment (miyamotok0105) [reply sources/layers/core.md]: rateに修正致しました。

File: sources/layers/core.md (hunk 8 above)
Comment (miyamotok0105) [reply sources/layers/core.md]: 入力と乗算されるバイナリドロップアウトマスクの型は1次元の整数テンソルで表す．に変更致しました。

File: sources/layers/core.md (hunk 9 above)
Comment (miyamotok0105) [reply sources/layers/core.md]: 補足的説明が思いついておりません。臨時的に平坦化に統一致しました。

File: sources/layers/core.md (hunk 10 above)
Comment (miyamotok0105) [reply sources/layers/core.md]: ','の位置を微調整致しました。

File: sources/layers/core.md (hunk 11 above)
Comment (miyamotok0105) [reply sources/layers/core.md]: 全角に変換致しました。

File: sources/layers/core.md (hunk 12 above)
Comment (miyamotok0105) [reply sources/layers/core.md]: 上記に習って修正致しました。

File: sources/layers/core.md (hunk 13 above)
Comment (miyamotok0105) [reply sources/layers/core.md]: 下流レイヤーに統一致しました。

File: sources/layers/core.md (hunk 8 above)
Comment (nzw0301) [reply sources/layers/core.md]: 型だとtypeと被るのでshapeでお願いします

Comment (miyamotok0105) [reply sources/layers/core.md]: convolutionalなど他の部分でも型という表現を使ってませんか？shapeに統一しますか？あと__ 引数__の部分が__Arguments__になったりバラツキが見えますがどっちに寄せますか？

Comment (nzw0301) [reply sources/layers/core.md]: 1. はい，使っています．（Convなどのマージがされてから表記揺れの議論が進んだため，そうなっています）
2. すでにpushされているページについては後で直します
3. `__引数__` でお願いします．

**Comment by nzw0301** [comment]
ping @miyamotok0105
